File: java/src/main/java/org/example/plus_one/PlusOne.java
package org.example.plus_one;

public class PlusOne {
    public static int[] plusOne(int[] digits) {
        int n = digits.length;
        int carry = 1;
        int[] res = new int[n];
        for (int i = n - 1; i >= 0; i--) {
            int sum = digits[i] + carry;
            res[i] = sum %10;
            carry = sum/10;
        }
        if (carry == 1) {
            res = new int[n+1];
            res[0] = 1;
        }
        return res;
    }
}

File: python/plus_one/plus_one.py

from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n = len(digits)
        carry = 1  # Initialize the carry to 1 since we want to increment by one.

        for i in range(n - 1, -1, -1):
            total = digits[i] + carry
            digits[i] = total % 10  # Update the current digit.
            carry = total // 10  # Calculate the carry for the next iteration.

        # If there's still a carry after the loop, insert it as the most significant digit.
        if carry:
            digits.insert(0, carry)

        return digits